packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.tsx

```diff
@@ -719,6 +724,25 @@ export class DetailsList extends BaseComponent<IDetailsListProps, IDetailsListSt
       onActiveItemChanged(items[index], index, ev);
     }
   }
+
+  private _getItemKey(item: any, itemIndex: number): string | number {
+    const { getKey } = this.props;
+
+    let itemKey: string | number | undefined = undefined;
+    if (item) {
+      itemKey = item.key;
```

Comment:
One thing that Christian and I were talking about a while ago is that we should consider using `id` rather than `key` for identifying items, since React doesn't like you passing in "key" as a prop to a thing.

I'm not sure it merits a change here but I would be OK if you agreed. We could support either.

Reply:
For the record (from Teams): That's a fair point, but I think we should address that in a separate PR? Would be a bigger change.. change the `IObjectWithKey` interface and update all consumers etc

Reply:
@cschleiden thanks so much for getting this in!!